S14-roles/generic-subtyping.t

```diff
@@ -0,0 +1,24 @@
+use v6;
+
+use Test;
+
+plan 5;
+
+lives-ok({ EVAL(Q:to/ROLES/) }, 'can do subtyped generic roles');
+role R1[Any ::T] { }
+role R2[Cool ::T] does R1[T] { }
+ROLES
+
+EVAL(Q:to/TESTS/);
```

Comment:
Is `EVAL` actually needed here?

Reply:
Wouldn't the file not compile if the test failed?

Reply:
A failed compilation is a failed test too. I just feel uneasy over doing tests inside `EVAL`s.

Reply:
Yeah, but a failed compilation could be caused by many things: I would consider a failed compilation an error in the test creation, rather than a failed test.

Reply:
I think the `EVAL` is needed here because the creation of the `R1` and `R2` roles are runtime, so referring to them here without `EVAL` would be a compile time error for the test-file.